ci/hack/rosa_cluster_provision.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+#quit if exit status of any cmd is a non-zero value
+set -o errexit
+set -o nounset
+set -o pipefail
+
+usage() {
+    echo "
+Usage: 
+    ${0##*/} ./rosa_cluster_provision.sh [options]
+
+Provision a ROSA cluster
+
+Mandatory arguments:
+    -t, --token
+        API token to authenticate against the Red Hat OpenShift Service on AWS account.
+    --id
+        AWS access key id.
+    --key
+        AWS secret access key.
+    -r, --region
+        AWS region name.
+    -n, --name
+        name of the cluster.
+    
+Optional arguments:
+    -d, --debug
+        Activate tracing/debug mode.
+    -h, --help
+        Display this message.
+Example:
+    ${0##*/} ./rosa_cluster_provision.sh --token <ROSA-API-TOKEN> --region us-west-2
+" >&2
+
+}
+
+parse_args() {
+  while [[ $# -gt 0 ]]; do
+    case $1 in
+    -t | --token)
+      shift
+      export ROSA_TOKEN="$1"
+      ;;
+    --id)
+      shift
+      export AWS_ACCESS_KEY_ID="$1"
+      ;;
+    --key)
+      shift
+      export AWS_SECRET_ACCESS_KEY="$1"
+      ;;
+    -r | --region)
+      shift
+      export AWS_REGION="$1"
+      ;;
+    -n | --name)
+      shift
+      export CLUSTER_NAME="$1"
+      ;;
+    -d | --debug)
+      set -x
+      ;;
+    -h | --help)
+      usage
+      exit 0
+      ;;
+    *)
+      echo "Unknown argument: $1"
+      usage
+      exit 1
+      ;;
+    esac
+      shift
+  done
+}
+
+init() {
+  tmpdir=$(mktemp -d)
+  cd "$tmpdir"  
+}
+
```

Comment:
You are not checking for any of the mandatory arguments.

Reply:
Added